src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -247,11 +247,47 @@ public AzureAppConfigurationOptions Connect(Uri endpoint, TokenCredential creden
             }
 
             ConnectionString = null;
-            Endpoint = endpoint;
+            Endpoints = new List<Uri>() { endpoint };
             Credential = credential;
             return this;
         }
 
+        /// <summary>
+        /// Connect the provider to Azure App Configuration and their replicas using list of endpoints and token credential.
+        /// </summary>
+        /// <param name="endpoints">The list of endpoints of the Azure App Configuration and it's replicas to connect to.</param>
+        /// <param name="credential">Token credential to use to connect.</param>
+        public AzureAppConfigurationOptions Connect(IEnumerable<Uri> endpoints, TokenCredential credential)
+        {
+            if (endpoints == null || endpoints.Count() == 0)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            var set = new HashSet<String>();
+
+            // Best effort to find the primary endpoint (the shortest one in length)
+            var shortestEndpoint = endpoints.OrderBy(e => e.Host.Length).First().Host;
+
+            // If the shortest endpoint is a replica or the store name has a '-', the common prefix would be string before first '-'
+            // else if there is no '-' in it, then it is not a replica and the common prefix would be a substring before first '.'
+            var commonPrefix = shortestEndpoint.Substring(0, shortestEndpoint.IndexOf('-') > 0 ? shortestEndpoint.IndexOf('-') : shortestEndpoint.IndexOf('.'));
+
+            if (!endpoints.Select(e => e.Host).All(set.Add) || // All endpoints must be unique.
+                !endpoints.All(e => e.Host.Substring(e.Host.IndexOf('.')).Equals( // All endpoints must have the same domain.
+                                    shortestEndpoint.Substring(shortestEndpoint.IndexOf('.')))) ||
```

Comment:
>shortestEndpoint.Substring(shortestEndpoint.IndexOf('.') [](http://example.com/codeflow?start=36&length=56)

We should extract the `commonDomain` beforehand instead of calculating it repeatedly in the LINQ.